geomstats/learning/exponential_barycenter.py

```diff
@@ -0,0 +1,192 @@
+"""Frechet mean."""
+
+import logging
+
+from sklearn.base import BaseEstimator
+
+import geomstats.backend as gs
+from geomstats.geometry.euclidean import Euclidean
+
+EPSILON = 1e-6
+
+
+def linear_mean(points, weights=None):
+    """Compute the weighted linear mean.
+
+    The linear mean is the Frechet mean when points:
+    - lie in a Euclidean space with Euclidean metric,
+    - lie in a Minkowski space with Minkowski metric.
+
+    Parameters
+    ----------
+    points : array-like, shape=[n_samples, dimension]
+        Points to be averaged.
+
+    weights : array-like, shape=[n_samples, 1], optional
+        Weights associated to the points.
+
+    Returns
+    -------
+    mean : array-like, shape=[1, dimension]
+        Weighted linear mean of the points.
+    """
+    if isinstance(points, list):
+        points = gs.vstack(points)
+    points = gs.to_ndarray(points, to_ndim=2)
+    n_points = gs.shape(points)[0]
+
+    if isinstance(weights, list):
+        weights = gs.vstack(weights)
+    elif weights is None:
+        weights = gs.ones((n_points, ))
+
+    weighted_points = gs.einsum('...,...j->...j', weights, points)
+    mean = (gs.sum(weighted_points, axis=0) / gs.sum(weights))
+    mean = gs.to_ndarray(mean, to_ndim=2)
+    return mean
+
+
+def _default_gradient_descent(
```

Comment:
Should we create a file with all the gradient descents, that would be used across learning algorithms, so that we do not have to copy-paste (this is copy-pasted from frechet_mean.py right?)

Reply:
Not exactly copy-paste but very close:
- different stopping criterion (which could be used in FrechetMean to make it faster maybe)
- factorizes `log(points, current)` to `log_from_id(compose(inverse(current), points))`

Reply:
- Maybe add a flag about the stopping criterion, and then factorize this _default_gradient_descent?
- Does this modification make the code faster? If so, we should probably compute all Lie group logs this way, and factorize _default_gradient_descent.

Reply:
This is already the way lie group logs are computed, and also in `invariant_metric` although I'm not sure it's right

Reply:
In that case, no need to change the code of `default_gradient_descent` and we can factorize it?